src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/DepartureReasonEntity.kt

```diff
@@ -34,9 +39,21 @@ data class DepartureReasonEntity(
   val isActive: Boolean,
   val serviceScope: String,
   val legacyDeliusReasonCode: String?,
+  @ManyToOne(fetch = FetchType.LAZY)
+  @JoinColumn(name = "parent_reason_id")
+  val parentReasonId: DepartureReasonEntity?,
 ) {
   override fun toString() = "DepartureReasonEntity:$id"
 
+  fun generateParentChildName(reason: DepartureReasonEntity?): NamedId? =
```

Comment:
i don't think we need to pass reason in here as you can directly access the fields you need (it's not longer a static function)